src/sink/gpub.rs

```diff
@@ -74,11 +78,13 @@ impl offramp::Impl for GoogleCloudPubSub {
                 .as_ref()
                 .ok_or("Offramp Google Cloud Pubsub (pub) requires a config")?,
         )?;
-        let remote = Some(block_on(pubsub_auth::setup_publisher_client())?);
+        let remote_publisher = Some(block_on(pubsub_auth::setup_publisher_client())?);
```

Comment:
block_on will block the entire thread, we really will want to avid that.

Reply:
I'm not sure what to replace it with :(

Reply:
We probably would want to move the logic to creat a connection into `init` it's the general pattern we use for most offramps. `from_config` creates a skeleton that isn't functional yet, `Sink::init` initializes that skeleton. That way we have control over when we initiate connections. `Sink::init` for that reason is also `async` so you can just use `.await` there :) - you can take a look at TCP, REST for an example of how it is done in those

Reply:
Yes! Fixed them all in the updated PR

Reply:
wooooh!